build/ci_utils/src/fs.rs

```diff
@@ -138,6 +138,12 @@ pub fn remove_symlink_dir_if_exists(path: impl AsRef<Path>) -> Result {
     }
 }
 
+/// Create a symlink, or remove and recreate it if it already exists.
+pub fn create_or_update_symlink(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result {
```

Comment:
Please rename it to `create_or_update_symlink_dir`. The directory symlinks and regular file symlinks are handled differently on Windows.